test/integration/consul-container/upgrade/healthcheck_test.go

```diff
@@ -0,0 +1,328 @@
+package consul_container
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+
+	consulCluster "github.com/hashicorp/consul/integration/consul-container/libs/consul-cluster"
+	consulNode "github.com/hashicorp/consul/integration/consul-container/libs/consul-node"
+
+	"github.com/hashicorp/consul/integration/consul-container/libs/utils"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+
+	"github.com/stretchr/testify/require"
+)
+
+var curImage = flag.String("uut-version", "local", "docker image to be used as UUT (unit under test)")
+var latestImage = flag.String("latest-version", "latest", "docker image to be used as latest")
+
+const retryTimeout = 10 * time.Second
+const retryFrequency = 500 * time.Millisecond
+
+// Test health check GRPC call using Current Clients and Latest GA Servers
+func TestLatestGAServersWithCurrentClients(t *testing.T) {
+	t.Parallel()
+	numServers := 3
+	Cluster, err := serversCluster(t, numServers, *latestImage)
```

Comment:
nit: only package-exported variables/constants/types should have names beginning with a capital letter.